pkg/target/target.go

```diff
@@ -595,73 +605,31 @@ func tplFuncMap() template.FuncMap {
 	return f
 }
 
-func processTemplateValues(valuesMap map[string]interface{}, templateContext map[string]interface{}) (map[string]interface{}, error) {
-	tplFn := template.New("values").Funcs(tplFuncMap()).Option("missingkey=error")
-	recursionDepth := 0
-	tplResult, err := templateSubstitutions(valuesMap, templateContext, tplFn, recursionDepth)
+func processTemplateValues(helmValues map[string]interface{}, templateContext map[string]interface{}) (map[string]interface{}, error) {
+	data, err := kyaml.Marshal(helmValues)
 	if err != nil {
-		return nil, err
+		return nil, fmt.Errorf("failed to unmarshal helm values template: %w", err)
 	}
-	compiledYaml, ok := tplResult.(map[string]interface{})
-	if !ok {
-		return nil, fmt.Errorf("templated result was expected to be map[string]interface{}, got %T", tplResult)
-	}
-
-	return compiledYaml, nil
-}
 
-func templateSubstitutions(src interface{}, templateContext map[string]interface{}, tplFn *template.Template, recursionDepth int) (interface{}, error) {
-	if recursionDepth > maxTemplateRecursionDepth {
-		return nil, fmt.Errorf("maximum recursion depth of %v exceeded for current templating operation, too many nested values", maxTemplateRecursionDepth)
+	tmpl := template.New("values").Funcs(tplFuncMap()).Option("missingkey=error").Delims("${", "}")
```

Comment:
I wonder if it's worth putting a comment ahead of this to let people know that `{}[],` are yaml flow control characters which causes issues with parsing `{{ }}` as valid YAML using the k8s parser we use. Just to explain the reason in a code comment if people wonder why.